src/types.ts

```diff
@@ -44,6 +44,91 @@ export type DefaultCategory =
   | 'NAVIGATION'
   | 'TABLE';
 
+interface RefOrValue {
+  type: 'THEME_COLOR' | 'STATIC';
+  value: string;
+}
+
+export interface StyleDefinitionCssObject {
+  backgroundColor?: RefOrValue;
+  borderColor?: RefOrValue;
+  borderRadius?: string[];
+  borderStyle?: string;
+  borderWidth?: string[];
+  boxShadow?: string;
+  color?: RefOrValue;
+  fontFamily?: string;
+  fontSize?: string;
+  fontStyle?: string;
+  fontWeight?: string;
+  letterSpacing?: string;
+  lineHeight?: string;
+  padding?: string[];
+  textDecoration?: string;
+  textTransform?: string;
+}
+export interface StyleDefinitionState {
+  name: StyleStateKeys;
+  cssObject: StyleDefinitionCssObject;
+}
+
+export interface StyleDefinition {
+  type: string;
+  name: string;
+  basis: StyleDefinitionCssObject;
+  states: StyleDefinitionState[];
+}
+
+export enum AllowedStateKeys {
+  SELECTED = 'selected',
+  HOVER = 'hover',
+  FOCUS = 'focus',
+  DISABLED = 'disabled',
+  VALID = 'valid',
+  INVALID = 'invalid',
+  READONLY = 'readOnly',
+}
+
+export type StyleStateKeys =
+  | 'selected'
+  | 'hover'
+  | 'focus'
+  | 'disabled'
+  | 'valid'
+  | 'invalid'
+  | 'readOnly';
+
+export type StyleDefinitionContentBase = {
+  [key in StyleStateKeys | 'basis']: StyleDefinitionCssObject;
+};
+
+export type StyleDefinitionContentKeys = {
+  [key in StyleStateKeys]?: string[];
+} & { basis: string[] };
+
+export interface StyleDefinitionContentOverwrites
+  extends Omit<StyleDefinitionState, 'name'> {
+  name: string;
+}
+
+export type BuildStyleDefinitionContentOverwrites =
+  Partial<StyleDefinitionContentBase>;
```

Comment:
Can you be more specific here? 😁

Reply:
the StyleDefinitionContentBase is specified to have all keys present in the object so this is as specific as we could be we want all the keys to be optional as the base requires them all to be present